src/components/HeaderBar.vue

```diff
@@ -0,0 +1,96 @@
+<script setup>
+import IconBaseball from '../components/icons/IconBaseball.vue';
+import IconBasketball from '../components/icons/IconBasketball.vue';
+import IconFootball from '../components/icons/IconFootball.vue';
+import IconHockey from '../components/icons/IconHockey.vue';
+import { ref } from 'vue';
+
+const props = defineProps({ allSports: { type: Array, required: true } })
+// let thisComponent = ref(IconBaseball)
+// ^^^tried using this to make the component :is dynamic, but will circle back to it
+</script>
+
+<template>
+  <div class="container">
+    <div class="sports-wrapper">
+      <button
+        v-for="sport in allSports" 
+        :key="sport.key"
+        class="sport"
+      >
+        <IconBaseball
+          v-if="sport.key === 'mlb'"
+          class="icon"
+        />
+      
+        <IconHockey
+          v-else-if="sport.key === 'nhl'" 
+          class="icon"
+        />
+      
+        <IconBasketball
+          v-else-if="sport.key === 'nba'" 
+          class="icon"
+        />
+      
+        <IconFootball
+          v-else-if="sport.key === 'nfl'" 
+          class="icon"
+        />
```

Comment:
Ok great.

Since there's no `v-else` that's going to be useful here as a default, these can just be `v-if`